### ISSUE TITLE
test/true help menu

### ISSUE THREAD

**Issue by dlegaultbbry** [landley/toybox#443]
We're trying to generate the 'docs' for the enabled commands and stumbled on the fact that the 2 commands below do not generate the help menu at all

toybox true --help
toybox test --help

Both have TOYFLAG_NOHELP being set on them, but should they have it?

**Comment by landley** [comment]
Yes, explicitly. You have to either "toybox --help true" or "help true" to get their help text. It should also show up in "help -ah" which is what generates https://landley.net/toybox/help.html

The commit adding this (e98089ab9) was in 2016 and I'm not finding the relevant discussion on the mailing list, and the second half of that year's blog was left unpublished due to the ITAR export regulations I hit doing a contract at innoflight and then I accidentally rsynced the web version over my local coly and I've got it on a backup disk somewhere but would have to dig through multiple boxes to figure out which one.

The discussion also might have been here on github, but microsoft has been borking github so badly that going to the closed issue tab and saying "sort by oldest" goes currently "2015, 2018, 2018, 2018, 2016, 2016" and when the website can no longer COUNT... well, that's microsoft for you.

Anyway, the underlying issue was that arguments to true and false are _explicitly_ ignored, and it turns out scripts depend on this. For example, I myself use the ": ${BLAH:-value} ${BLAH:-value} ${BLAH:-value}" trick where ":" is an alias for true and "${VAR:-value}" is the bash syntax to assign a default value to the variable you're evaluating if it's empty (the colon means trigger when unset or set to "", without that the - assign-default-value operator only triggers when the variable is unset). That's a quick way to do multiple [ -n "$BLAH" ] && BLAH=value on the same line, but if the first already inherited the value of "--help" or "--version" it would resolve to that and trigger behavior. This also came up with the "test" command where test "$X" == "$Y" was breaking when $X resolved to "--version"...

And thus NOHELP was born. Because sometimes, a command does NOT want to parse the "standard" arguments...

**Comment by dlegaultbbry** [comment]
The justification makes sense! I'll use `toybox --help <cmd>` to generate the help which works with all of them. That's good enough for what I need!

**Comment by enh-google** [comment]
> The discussion also might have been here on github, but microsoft has been borking github so badly that going to the closed issue tab and saying "sort by oldest" goes currently "2015, 2018, 2018, 2018, 2016, 2016" and when the website can no longer COUNT... well, that's microsoft for you.

("oldest" means creation date. "recently updated" is modification date. but, yeah, we're programmers --- they could just use those words!)
